### ISSUE TITLE
Questions about the models

### ISSUE THREAD

**Issue by rainskyfyy** [Alina9/distillation-vilt#3]
Hi, thank you for releasing such a good project. I have several questions during running your code. Could you help me, please?
First, I'm wondering if your code is doing whole pre-train task during the train stage, or is just finetuning a pre-trained model?
Second, what's the structure of your student model?(it should be a vilt model right?) and where can I get it?
So sorry about bothering you here. I would appreciate it if you could reply me. Thank you so much!

**Comment by Alina9** [comment]
Hi!
The architecture of the model coincided with the model  ViLT from the article (https://arxiv.org/abs/2102.03334).

Here is my diploma https://disk.yandex.ru/i/XWbYM5bRBZpNkQ. It is written in Russian. But you can find illustrations of the model distillation-vilt.
My model uses pre-train weights of BERT and ViT models. This weights uses for distillation. Then I train distillation model. Unfortunately, I can train model only on small dataset MS COCO. I show that my model with distillation reaches beter result than without distillation. 
Also I don't use  AdamW as optimizer, I think it is my mistake, because ViLT was trained with this one.

Please, show me results, if you try to train model on big dataset or other optimizer.

**Comment by rainskyfyy** [comment]
Sorry for replying so late. I believe we have different goals of distillation here. You try to distillate the BERT model and a ViT model to get a ViLT model, but my purpose is to distillate a ViLT model and compress it to a smaller size, which means the teacher is a ViLT as well as the student. The only difference between them is the number of transfomer layers (teacher has 12 and student has 6).
Still, thank you so much cause I just built my code on the basis of yours. And the dataset I use is also MS COCO. I'm sorry I have no result to show you from a bigger dataset. As for the optimizer, I tried with AdamW as you told me. I'm still waiting for the outcome since the training is really slow (I'm training on a Tesla V100-PCIE-32GB, but every batch (batch_size=64) takes at least 7 seconds, I'm wondering if I have made anything wrong). I will show you the details when it's over. 
Thank you again! Best wishes!
